virtcontainers/network.go

```diff
@@ -995,6 +1007,47 @@ func bridgeNetworkPair(netPair *NetworkInterfacePair, numCPUs uint32, disableVho
 	return nil
 }
 
+func tuntapNetworkPair(endpoint *VirtualEndpoint, numCPUs uint32, disableVhostNet bool) error {
```

Comment:
Why are we doing things differently than `tapNetworkPair()`? If you're trying to tie an interface to the VM, you need this interface to be connected to the TAP interface being created, right?
I don't get why the model implemented by `tapNetworkPair()`, and relying on MACVTAP cannot be used for this use case?

Reply:
We just create a tap device but no veth pair. So I create a new function. Maybe function name is confusing. It can reuse the tapNetworkPair() with some refactoring.